client/session.go

```diff
@@ -1370,10 +1401,10 @@ func (s *session) FetchBlocksFromPeers(
 
 	// Begin consuming metadata and making requests
 	go func() {
-		err := s.streamBlocksFromPeers(nsMetadata, shard, peers,
-			metadataCh, opts, result, m, s.passThruBlocksMetadata)
+		s.streamBlocksFromPeers(nsMetadata, shard, peers,
+			metadataCh, opts, result, progress, s.passThruBlocksMetadata)
 		close(outputCh)
-		onDone(err)
+		onDone(nil)
```

Comment:
Hm why don't we return error here too?

Reply:
same comment as above

Reply:
Right, yeah so perhaps we should change the `onDone(...)` to just be `func()` instead of `func(error)` because it seems like this is the only caller of `onDone` in this method?

Reply:
I can do this, the only problem is that its passed into the peerBlocksIter which has this signature:

```golang
func newPeerBlocksIter(
	inputC <-chan peerBlocksDatapoint,
	errC <-chan error,
) *peerBlocksIter {
	return &peerBlocksIter{
		inputCh: inputC,
		errCh:   errC,
	}
}

func (it *peerBlocksIter) Current() (topology.Host, ts.ID, block.DatabaseBlock) {
	return it.current.peer, it.current.id, it.current.block
}

func (it *peerBlocksIter) Err() error {
	return it.err
}

func (it *peerBlocksIter) Next() bool {
	if it.done || it.err != nil {
		return false
	}
	m, more := <-it.inputCh

	if !more {
		it.err = <-it.errCh
		it.done = true
		return false
	}

	it.current = m
	return true
}
```

I can re-write the iter to never return an err / not accept an errCh but its a bigger change.

What do you think?

Reply:
Spoke in person. We settled on leaving the onDone function signature as is because we want to leave open the possibility of needed to return an error from the peerBlocksIter interface in the future